README.md

```diff
@@ -0,0 +1,317 @@
+> Amazon’s S3 strata with the icy calm of a hyper-focused courier, slices
+> latency like fresh sashimi, and tattoos every object with cryptographic
+> sigils before the competition has even parsed their credentials. Because in
+> this fractured mesh of micro-services and franchised LLMs, sovereignty
+> belongs to whoever controls the pipeline—and s3nd is the midnight-black
+> muscle car that shows up, engine snarling, ready to haul your binary payload
+> through rush-hour congestion in under thirty seconds, refund guaranteed,
+> pride irrevocable.
+>
+> -- Hiro Photogonist, Latency Survey of S3 Transfers
+
+## Problem Space
+
+The transfer of data across high latency (> 200ms) long-haul networks where the
+data is produced in periodic bursts. Each burst is composed of multiple related
+files and the total transfer time for all files within the burst must be
+minimized.
+
+## Architecture
+
+The Deliverator design is based on [`lsst-dm/s3daemon`](https://github.com/lsst-dm/s3daemon).
+
+A pool of cached https connection is maintained.
+Transfers are aggressively timed out and retried as trade off between efficient use of bandwidth and minimizing the completion time of a file transfer.
+
+## Upload Request Submission
+
+Local files are submitted for upload to a `s3nd` instance via HTTP POST request.
+All parameters are `application/x-www-form-urlencoded`, which makes it simple to create a upload request with common tools such as `curl`.
+
+There are two mandatory parameters to make a valid request:
+
+- `file`
+    The fully qualified path to the file to be uploaded.
+    The file (and its parent directories) must be accessible to the euid/egid the `s3nd` service is running as.
+    This parameter is required.
+- `uri`
+    The URI of the S3 object the file should be uploaded too.
+    The destination bucket must already exist.
+    Note that there is no checking if the object already exists.
+    If the s3 credentials in use by `s3nd` allow an object to be overwritten, `s3nd` will do so.
+    This parameter is required.
+
+Example of submitting a file for transfer with `curl`:
+
+```bash
+curl -d "file=/tmp/foo" -d "uri=s3://bar/foo" http://localhost:15555
+```
+
+The HTTP connection will be held open until by the `s3nd` server until a definitive transfer status (success or error) is returned via HTTP status code.
+The Deliverator does not maintain persistent state for queued or in progress uploads.
+If the service is restarted, all in progress transfers are abandoned.
+
+> [!IMPORTANT]
+> If an HTTP client of `s3nd` disconnects, I.e., the socket is closed, the request is considered abandoned.
+> If there is an s3 upload in progress, it will be aborted.
+
+A status message shall always be returned to the client encoded as `application/json`. A `code` with an http status code as integer value and `msg` string will be present.
+Other information may be part of the response.
+
+## Example Status Responses
+
```

Comment:
It would be cool if there were a Go framework that would handle this sort of documentation automagically the way Pydantic and FastAPI do, but I don't know if there is.

Reply:
Using `swaggo/swag` a `/swagger` endpoint has been added and `docs/swagger.md` has been committed to this repo and linked from the `README.md`.